### ISSUE TITLE
HTML formatter trims whitespace within span

### ISSUE THREAD

**Issue by jaimeiniesta** [phoenixframework/phoenix_live_view#1985]
### Environment

* Elixir version (elixir -v): 1.13.4
* Phoenix version (mix deps): 1.6.7
* Phoenix LiveView version (mix deps): 0.17.9
* Operating system: macOS Monterey
* Browsers you attempted to reproduce this bug on (the more the merrier): n/a
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: n/a

### Actual behavior

Running `mix format` is trimming whitespace within `<span>` elements.

### Expected behavior

Whitespace not being trimmed.

I'm using Tailwind to show shorter or longer texts in some places depending on the space available. For example, in the pricing page where the user can choose the plan, we can have:

* weekly / monthly / yearly (for mobile)
* weekly billing / monthly billing / yearly billing (for desktop)

To achieve this, we use this:

```html
<p>Weekly<span class="hidden md:inline"> billing</span></p>
```

That is, `Weekly` is always shown, and the ` billing` part within the `span` (that includes a leading space) is hidden by default, and inline on larger breakpoints.

This works fine, but after running `mix format` with the new HEEx formatter enabled, it removed the whitespace within the `span`,  which I didn't expect. Maybe there's a reason for that, but that's not what I'd expect. Please correct me if I'm wrong.

In short, this:

```html
<span> something </span>
```

will be converted to:

```html
<span>something</span>
```

while this:

```html
<span>some       thing</span>
```

will not be converted to

```html
<span>some thing</span>
```

which could make more sense to me.

As a workaround, I used `&nbsp;` in place of the leading whitespace.

**Comment by leandrocp** [comment]
Hi @jaimeiniesta, have you tested on master? There are some unreleased changes that my have impact in your case, please give it a try, that's the current behavior:

```
iex> IO.puts Phoenix.LiveView.HTMLFormatter.format("<p>Weekly<span class=\"hidden md:inline\"> billing</span></p>", [])
<p>
  Weekly<span class="hidden md:inline"> billing</span>
</p>
```

And for special cases, to avoid formatting you can mark the element as `phx-no-format`, see https://github.com/phoenixframework/phoenix_live_view/pull/1977 and https://github.com/phoenixframework/phoenix_live_view/pull/1980

**Comment by feliperenan** [comment]
Would it work if you add a space between text and span instead of having it within the tag? I mean something like this:

```html
<p>Weekly <span class="hidden md:inline">billing</span></p>
```

The formatter only trims whitespace as you said. It doesn't remove text between texts though and I'm not sure if it should 🤔

**Comment by feliperenan** [comment]
But yes, the main issue seems to be fixed on master. Let me know otherwise 🙌🏻

**Comment by jaimeiniesta** [comment]
Thank you, I've tested on master and the results are a bit inconsistent regarding new lines, but it respects whitespace:

```elixir
iex(17)> IO.puts Phoenix.LiveView.HTMLFormatter.format("<h1>Sign up<span> now</span></h1>", [])                  
<h1>      
  Sign up<span> now</span>
</h1>

iex(18)> IO.puts Phoenix.LiveView.HTMLFormatter.format("<h1>Sign up <span> now</span></h1>", [])
<h1>Sign up <span>now</span></h1>
```

**Comment by 03juan** [comment]
@jaimeiniesta the second example seems to be applying the 4th step of the [collapsible-white-space](https://www.w3.org/TR/css-text-3/#collapsible-white-space) CSS3 rule, so technically working as intended.

>Any [collapsible](https://www.w3.org/TR/css-text-3/#collapsible-white-space) [space](https://www.w3.org/TR/css-text-3/#spaces) immediately following another collapsible space—even one outside the boundary of the inline containing that space, provided both spaces are within the same inline formatting context—is collapsed to have zero advance width. (It is invisible, but retains its [soft wrap opportunity](https://www.w3.org/TR/css-text-3/#soft-wrap-opportunity), if any.)

I would argue that the formatter should preserve the line breaks of inline elements and not insert breaks like the first example, and maybe at most collapse multiple consecutive spaces, such that the browser is able to retain the soft wrap opportunity that I assume is prevented by removing the space after <span> in the second example.
